File: src/casters/dto/create-caster.dto.ts
import {IsEmail, IsString } from "class-validator";
import {ApiProperty} from "@nestjs/swagger";

export class CreateCasterDto {
    @ApiProperty()
    @IsEmail()
    email: string;

    @ApiProperty()
    @IsString()
    password: string;

    @ApiProperty()
    @IsString()
    gamerTag: string;
}

File: riv4lz-api/riv4lz.casterApi/Dtos/CreateCasterDto.cs
using System.ComponentModel.DataAnnotations;

namespace riv4lz.casterApi.Dtos;

public class CreateCasterDto
{
    [Required] public string Email { get; set; }
    [Required] public string GamerTag { get; set; }
    [Required] public string Password { get; set; }
}